### DISCUSSION TITLE
Hi - I'm new to the community. I 've started using the library which is very useful. I have a similar question to the above post: In my case I want to set the time from a variable. Something like:

### DISCUSSION THREAD

**Discussion by Makuna** [Makuna/Rtc#133] [Q&A] [answered]
(@RussellSher I moved this question out of the welcome discussion into the QA area where it belongs, not an easy thing to do)

Hi - I'm new to the community. I 've started using the library which is very useful. I have a similar question to the above post: In my case I want to set the time from a variable. Something like:
```
void setDs1302()
{
  char datestring[20];
  char setmonth = 9;
  char setday = 4;
  int setyear = 2021;
  char sethour = 8;
  char setmin = 23;
  char setsec = 0; 
  snprintf_P(datestring,
             countof(datestring),
             PSTR("%04u/%02u/%02u %02u:%02u:%02u"),
             setyear,
             setmonth,
             setday,
             sethour,
             setmin,
             setsec );

Serial.println(datestring);
Serial.println("setting...");
Rtc.SetDateTime(datestring);
}
```
but it doesn't update the ds1302 - the 1302 is fine and the example code works. I am obviously not using the RtcDateTime object argument correctly. Could you advise please? regards
Russell

_Originally posted by @RussellSher in https://github.com/Makuna/Rtc/discussions/121#discussioncomment-1674596_

**Comment by Makuna** [comment, marked as answer]
@RussellSher First, your example doesn't even compile.  if you look at the Wiki for RTC, there is no SetDateTime that takes a string; it takes a RtcDateTime object.  

If you then look at this [Wiki entry on the constructors for RtcDateTime](https://github.com/Makuna/Rtc/wiki/RtcDateTime-object#constructors) you will notice that RtcDateTime can be initialized without using a string at all.  

Review the examples, they demonstrate the creation of a RtcDateTime object and just construct one using the `RtcDateTime(uint16_t year, uint8_t month, uint8_t dayOfMonth, uint8_t hour, uint8_t minute, uint8_t second)` constructor.

**Reply by RussellSher** [reply]
Thanks- (sorry for posting there, it was a bit confusing with telling me to introduce myself before posting)
This does the trick, is this the idea:
` RtcDateTime manual = RtcDateTime("Nov 20 2021", "19:55:00");
Rtc.SetDateTime(manual);`
It does do the job for me.
from here: [https://forum.arduino.cc/t/solved-ds3231-set-date-time-manually/454709](url)

**Comment by Makuna** [comment]
Why use strings at all?  From your example, you have the year, month, day of month, etc already.  Don't convert to a string, just supply them when you construct the RtcDateTime.
```
RtcDateTime manual = RtcDateTime(setyear, setmonth, setday, sethour, setminute, setsecond);
Rtc.SetDateTime(manual);
```

**Reply by RussellSher** [reply]
HI - thanks for that. Yes, that was much better.
Regards
Russell
